packages/shared/src/components/modals/user/CookieConsentModal.tsx

```diff
@@ -0,0 +1,128 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+import type { ModalProps } from '../common/Modal';
+import { Modal } from '../common/Modal';
+import { Typography, TypographyTag } from '../../typography/Typography';
+import { Divider } from '../../utilities';
+import { cookiePolicy } from '../../../lib/constants';
+import type { AcceptCookiesCallback } from '../../../hooks/useCookieBanner';
+import {
+  GdprConsentKey,
+  otherGdprConsents,
+} from '../../../hooks/useCookieBanner';
+import { Button, ButtonSize, ButtonVariant } from '../../buttons/Button';
+import { CookieConsentItem } from './CookieConsentItem';
+
+interface CookieConsentModalProps extends ModalProps {
+  onAcceptCookies: AcceptCookiesCallback;
+}
+
+const options = Object.values(GdprConsentKey).filter(
+  (key) => key !== GdprConsentKey.Necessary,
+);
+
+export const CookieConsentModal = ({
+  onAcceptCookies,
+  ...modalProps
+}: CookieConsentModalProps): ReactElement => {
+  const { onRequestClose } = modalProps;
+
+  const onAcceptPreferences = (e: React.FormEvent) => {
+    e.preventDefault();
+    // get the form
+    const formData = new FormData((e.target as HTMLInputElement).form);
+    const formProps = Object.fromEntries(formData);
+    const keys = Object.keys(formProps);
+    const acceptedConsents = keys.filter(
+      (key) =>
+        formProps[key] === 'on' &&
+        otherGdprConsents.includes(key as GdprConsentKey),
+    );
+    const rejectedConsents = options.filter(
+      (option) => !acceptedConsents.includes(option),
+    );
+    onAcceptCookies(acceptedConsents, rejectedConsents);
+
+    onRequestClose(null);
+  };
+
+  const onAcceptAll = (e: React.MouseEvent) => {
+    onAcceptCookies(otherGdprConsents);
+    onRequestClose(e);
+  };
+
+  const onRejectAll = (e: React.MouseEvent) => {
+    onAcceptCookies(); // this will accept just the necessary ones
+    onRequestClose(e);
+  };
+
+  return (
+    <Modal
+      {...modalProps}
+      kind={Modal.Kind.FlexibleCenter}
+      className="min-h-[38.75rem]"
+    >
+      <Modal.Header title="Cookie preferences" />
+      <Modal.Body>
+        <Typography type={Typography.Type.Body} bold>
```

Comment:
Think the imports got a bit weird here.
Shouldn't it be `TypographyType.Body`? And for all below

Reply:
Yeah, I updated the `Typography` to be similar to our Modal component where we can use everything from the `Typography` object alone.

Reply:
Ah not the biggest fan, wonder if it imports the whole thing then vs only 1 each?
But modal we still do ModalType instead of Modal.Type right?

Reply:
> Ah not the biggest fan, wonder if it imports the whole thing then vs only 1 each?

Haven't checked as I just followed the convention within the Modal.

> But modal we still do ModalType instead of Modal.Type right?

I think you are referring to the LazyModal.Type which is from a separate object. We can revert back to the original one, but should we then make the same thing for Modals?

Reply:
Ideally yes, but non blocking.
Just seems less of a import duty